pkg/descheduler/node/node.go

```diff
@@ -32,26 +32,32 @@ import (
 // ReadyNodes returns ready nodes irrespective of whether they are
 // schedulable or not.
 func ReadyNodes(client clientset.Interface, nodeSelector string, stopChannel <-chan struct{}) ([]*v1.Node, error) {
-	nl := GetNodeLister(client, stopChannel)
-
 	ns, err := labels.Parse(nodeSelector)
 	if err != nil {
 		return []*v1.Node{}, err
 	}
 
-	nodes, err := nl.List(ns)
-
-	if err != nil {
-		return []*v1.Node{}, err
+	var nodes []*v1.Node
+	nl := GetNodeLister(client, stopChannel)
+	if nl != nil {
+		// err is defined above
+		if nodes, err = nl.List(ns); err != nil {
+			return []*v1.Node{}, err
+		}
 	}
 
 	if len(nodes) == 0 {
-		var err error
+		glog.V(2).Infof("node lister returned empty list, now fetch directly")
```

Comment:
Any specific reason for choosing V(2)? I thought, we have fixed on 1 & 0 as levels.